bazel/grpc_python_deps.bzl

```diff
@@ -49,8 +49,10 @@ def grpc_python_deps():
     if "io_bazel_rules_python" not in native.existing_rules():
         http_archive(
             name = "io_bazel_rules_python",
-            url = "https://github.com/bazelbuild/rules_python/releases/download/0.0.1/rules_python-0.0.1.tar.gz",
-            sha256 = "aa96a691d3a8177f3215b14b0edc9641787abaaa30363a080165d06ab65e1161",
+            url = "https://github.com/bazelbuild/rules_python/releases/download/0.4.0/rules_python-0.4.0.tar.gz",
+            sha256 = "954aa89b491be4a083304a2cb838019c8b8c3720a7abb9c4cb81ac7a24230cea",
+            patches = ["//third_party:rules_python.patch"],
+            patch_args = ["-p1"],
         )
```

Comment:
We may want to add a short comment here to remind people that we need this patch for the RBE environment. Do you think we should create a bug to remind us that we need to update the RBE images?

Reply:
Even if we upgrade the RBE environments, I think this patch is still necessary. We need it for environments that don't have Python installed at all (like Windows RBE) and the minimum required version of `rules_python` is a moving target. We might update our RBE workers, but then find that the newest version of `rules_python` requires a version of Python _newer_ than what we have installed on our infrastructure.

I think the long term action item here is actually to try to upstream this patch to `rules_python`. But it'd call that P2. Since this dependency is only in our `WORKSPACE` and not in `grpc_python_deps`, there's no chance of it affecting those who use our repo as a Bazel dependency.

Reply:
In an offline discussion, we agree that upstream is the right approach here. Richard mentioned the `rules_python` repo should allow Python targets to exist in none-Python builds instead of hard failures.

Reply:
[Issue filed on the `rules_python` repo](https://github.com/bazelbuild/rules_python/issues/542) to ask about upstreaming.